server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryFactory.java

```diff
@@ -37,17 +39,25 @@ public RemoteSegmentStoreDirectoryFactory(Supplier<RepositoriesService> reposito
 
     @Override
     public Directory newDirectory(String repositoryName, IndexSettings indexSettings, ShardPath path) throws IOException {
+        String indexUUID = indexSettings.getIndex().getUUID();
+        String shardId = String.valueOf(path.getShardId().getId());
+        return newDirectory(repositoryName, indexUUID, shardId);
+    }
+
+    public Directory newDirectory(String repositoryName, String indexUUID, String shardId) throws IOException {
         try (Repository repository = repositoriesService.get().repository(repositoryName)) {
             assert repository instanceof BlobStoreRepository : "repository should be instance of BlobStoreRepository";
             BlobPath commonBlobPath = ((BlobStoreRepository) repository).basePath();
-            commonBlobPath = commonBlobPath.add(indexSettings.getIndex().getUUID())
-                .add(String.valueOf(path.getShardId().getId()))
+            commonBlobPath = commonBlobPath.add(indexUUID)
+                .add(shardId)
                 .add("segments");
 
             RemoteDirectory dataDirectory = createRemoteDirectory(repository, commonBlobPath, "data");
-            RemoteDirectory metadataDirectory = createRemoteDirectory(repository, commonBlobPath, "metadata");
+            RemoteDirectory metadataDirectory = createRemoteDirectory(repository, commonBlobPath,"metadata");
+            RemoteStoreMDShardLockManager mdLockManager = new RemoteStoreMDLockManagerFactory(repositoriesService)
+                .newLockManager(repositoryName, indexUUID, shardId);
```

Comment:
The reason i was creating factory instance is, there can be scenarios where we can use the same factory instance to create lock manager for different indices or shards. for ex, in case of snapshots, maybe we can just pass the factory instance from snapshot service to BlobStoreRepository instance and use that factory instance to create lock manager for different indices and shards.